SalesforceCommerceCloud/pwa-kit
# @W-14146843@ modularize country code source for targeting via extensibility

# Description

Currently the `useAddressFields.jsx` has a hardcoded value for the `options` dropdown. This means that you can't expand the country options without overriding the whole file (a pattern we want to avoid for making better use of the Template Extensibility feature!) 

This request comes from the Adyen team who need to be able to extend this file without overriding this whole file in their accelerator (which would be bad!) 

Note that we want to make small changes like this iteratively and not be shy about gradually "decomposing" the app code structure gradually over time

Closes #1408



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Run retail react app locally 
- Make a change to the `address-fields-country-codes.jsx` country list
- Refresh the page http://localhost:3000/global/en-GB/checkout
- Observe the updated list in accordance with your change

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
